### PR TITLE
Split from position

### LINKED ISSUE #14: Split positions from existing position
- Manually enter position id
- Validate position existence
- Validate user's position balance
- Use position's collateral as `collateralToken`
- Use positions's collection as `parentCollectionId`
- Use trivial partition

### PR DESCRIPTION
**Pull request by hpmaxi**
Closes #14

### CODE REVIEW

File: src/pages/split-condition/SplitCondition.tsx [hunk 1]
```diff
@@ -1,90 +1,142 @@
...
+import { InputAmount } from './InputAmount'
+import { InputPosition } from './InputPosition'
+import { SelectCollateral } from './SelectCollateral'
+import { InputCondition } from './InputCondition'
+import { fetchPosition_position } from 'types/generatedGQL'
+import { ERC20Service } from 'services/erc20'
+import { useWeb3Connected } from 'contexts/Web3Context'
+
+export const bytesRegex = /^0x[a-fA-F0-9]{64}$/
+export const NULL_PARENT_ID = '0x0000000000000000000000000000000000000000000000000000000000000000'
```
Comment (mariano-aguero) [inline src/pages/split-condition/SplitCondition.tsx:18]: I think this should exist in the constant file (Line 17 and 18)

File: src/pages/split-condition/SplitCondition.tsx [hunk 2]
```diff
@@ -1,90 +1,142 @@
...
+      } else {
+        throw Error('Invalid split origin')
+      }
+
+      reset({
+        amount: ZERO_BN,
+        collateral: tokens[0].address,
+        conditionId: '',
+        splitFrom: 'collateral',
+        positionId: '',
```
Comment (mariano-aguero) [inline src/pages/split-condition/SplitCondition.tsx:104]: This object is the same used in the default above in the line 54

File: src/types/generatedGQL.ts [hunk 3]
```diff
@@ -25,6 +25,39 @@ export interface Conditions {
 // @generated
 // This file was automatically generated and should not be edited.
 
+// ====================================================
```
Comment (mariano-aguero) [inline src/types/generatedGQL.ts:28]: Please run prettier in this file

File: src/util/tools.ts [hunk 4]
```diff
@@ -18,3 +19,14 @@ export const isContract = async (provider: Provider, address: string): Promise<b
 export const range = (max: number): number[] => {
   return Array.from(new Array(max), (_, i) => i)
 }
+
+export const trivialPartition = (size: number) => {
```
Comment (mariano-aguero) [inline src/util/tools.ts:23]: Can you add an small description to this method? What is the purpose? what is it for?

File: src/services/conditionalTokens.ts [hunk 5]
```diff
@@ -49,6 +54,17 @@ export class ConditionalTokensService {
     }
   }
 
+  static getCollectionId(parentCollection: string, conditionId: string, indexSet: BigNumber) {
+    return CTHelpers.combineCollectionIds([
+      parentCollection,
+      CTHelpers.getCollectionId(conditionId, indexSet),
```
Comment (mariano-aguero) [inline src/services/conditionalTokens.ts:60]: The `getCollectionId` receive 3 parameters, Am I missing something here? 

`function getCollectionId (bytes32 parentCollectionId, bytes32 conditionId, uint indexSet)externalviewreturns(bytes32)`

See https://docs.gnosis.io/conditionaltokens/docs/devguide03/

**Comment by mariano-aguero** [comment]
Error adding a wrong value in the position input
Peek 14-07-2020 14-02

File: src/pages/split-condition/InputAmount.tsx [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+
+      <Controller
+        name="amount"
+        rules={{ required: true, validate: (amount) => amount.gt(ZERO_BN) }}
+        control={control}
+        decimals={collateral.decimals}
+        as={BigNumberInputWrapper}
+      />
+      {balance && (
+        <button onClick={() => setValue('amount', balance)}>{`Wallet Balance ${formatBigNumber(
```
Comment (mariano-aguero) [inline src/pages/split-condition/InputAmount.tsx:68]: Maybe the text `Use wallet balance ...` is better than `Wallet balance`
05- CTEAF Split Position

File: src/pages/split-condition/SplitCondition.tsx [hunk 7]
```diff
@@ -141,18 +176,15 @@ export const SplitCondition = ({
...
-
-      <button type="submit" disabled={!canSubmit}>
+      <InputAmount
+        collateral={collateralToken}
+        positionId={positionId}
+        splitFrom={splitFrom}
+        formMethods={formMethods}
+      ></InputAmount>
+      <button onClick={handleSubmit(onSubmit)} disabled={!canSubmit}>
         Split
```
Comment (mariano-aguero) [inline src/pages/split-condition/SplitCondition.tsx:186]: If  the transaction is running, we could change the text to `Splitting...` or something like that, so the user knows what is happening.

File: src/types/generatedGQL.ts (hunk 3 above)
Comment (mariano-aguero) [reply src/types/generatedGQL.ts:28]: Maybe we can add some kind of husky hook , so we are sure that prettier was executed

File: src/pages/split-condition/SplitCondition.tsx (hunk 1 above)
Comment (hpmaxi) [reply src/pages/split-condition/SplitCondition.tsx:18]: This is done in your PR of condition details, should be solved after merge.

File: src/services/conditionalTokens.ts (hunk 5 above)
Comment (hpmaxi) [reply src/services/conditionalTokens.ts:60]: I think this was more general for nested conditions, and we can use CTHelpers off-chain provided by gnosis.

File: src/types/generatedGQL.ts (hunk 3 above)
Comment (mariano-aguero) [reply src/types/generatedGQL.ts:28]: We will ignore this file in the eslint/prettier configuration

**Pull request merged**